### ISSUE TITLE
Postmark multiple recipients fails

### ISSUE THREAD

**Issue by kareemcoding** [anymail/django-anymail#227]
**Anymail version:** 8.1
**ESP:** Postmark

**Your ANYMAIL settings:**
```
ANYMAIL = {
     "POSTMARK_SERVER_TOKEN": os.environ['POSTMARK_SERVER_TOKEN']
}

EMAIL_BACKEND = "anymail.backends.postmark.EmailBackend"
```

**Versions of Django, requests, python:**

- `requests==2.24.0`
- `Django==2.2.16`
- `Python 3.7.3`

**Exact error message and/or stack trace**
```
Environment:


Request Method: POST
Request URL: http://localhost:8000/client/53024fd2-bdfb-4714-9e6e-e725e7512aca/changes/comment/88712cd7-7427-4eb1-a6ce-9aa75c09b1d4/

Django Version: 2.2.16
Python Version: 3.7.3
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.humanize',
 'internal',
 'clients',
 'user',
 'anymail',
 'crispy_forms',
 'storages',
 'social_django',
 'polymorphic']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'whitenoise.middleware.WhiteNoiseMiddleware']



Traceback:

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/django/core/handlers/exception.py" in inner
  34.             response = get_response(request)

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/django/core/handlers/base.py" in _get_response
  115.                 response = self.process_exception_by_middleware(e, request)

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/django/core/handlers/base.py" in _get_response
  113.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/sentry_sdk/integrations/django/views.py" in sentry_wrapped_callback
  36.                     return callback(request, *args, **kwargs)

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/django/views/generic/base.py" in view
  71.             return self.dispatch(request, *args, **kwargs)

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/django/utils/decorators.py" in _wrapper
  45.         return bound_method(*args, **kwargs)

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/django/contrib/auth/decorators.py" in _wrapped_view
  21.                 return view_func(request, *args, **kwargs)

File "/Users/chesampat/Documents/GitHub/iota/clients/generics.py" in dispatch
  22.             return super().dispatch(request, *args, **kwargs)

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/django/views/generic/base.py" in dispatch
  97.         return handler(request, *args, **kwargs)

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/django/views/generic/edit.py" in post
  142.             return self.form_valid(form)

File "/Users/chesampat/Documents/GitHub/iota/clients/views.py" in form_valid
  340.         comment.save()

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/django/db/models/base.py" in save
  744.                        force_update=force_update, update_fields=update_fields)

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/django/db/models/base.py" in save_base
  793.                 update_fields=update_fields, raw=raw, using=using,

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/django/dispatch/dispatcher.py" in send
  175.             for receiver in self._live_receivers(sender)

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/django/dispatch/dispatcher.py" in <listcomp>
  175.             for receiver in self._live_receivers(sender)

File "/Users/chesampat/Documents/GitHub/iota/clients/signals.py" in send_comment_notification
  19.     send.comment_notification(instance.change_request.client, instance)

File "/Users/chesampat/Documents/GitHub/iota/clients/emails.py" in comment_notification
  67.     send_email(to_emails, template_id, data)

File "/Users/chesampat/Documents/GitHub/iota/src/utils.py" in send_email
  14.     message.send()

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/django/core/mail/message.py" in send
  306.         return self.get_connection(fail_silently).send_messages([self])

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/anymail/backends/base.py" in send_messages
  94.                     sent = self._send(message)

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/anymail/backends/base_requests.py" in _send
  58.         return super()._send(message)

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/anymail/backends/base.py" in _send
  124.         response = self.post_to_esp(payload, message)

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/anymail/backends/base_requests.py" in post_to_esp
  69.         params = payload.get_request_params(self.api_url)

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/anymail/backends/postmark.py" in get_request_params
  177.         params = super().get_request_params(api_url)

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/anymail/backends/base_requests.py" in get_request_params
  165.             data=self.serialize_data(),

File "/Users/chesampat/Documents/GitHub/iota/env/lib/python3.7/site-packages/anymail/backends/postmark.py" in serialize_data
  190.             assert len(self.to_emails) == 1

Exception Type: AssertionError at /client/53024fd2-bdfb-4714-9e6e-e725e7512aca/changes/comment/88712cd7-7427-4eb1-a6ce-9aa75c09b1d4/
Exception Value: 

```

The errors looks to be happening when multiple to_emails are provided to the Anymail postmark backend.

This `assert len(self.to_emails) == 1` seems to be causing the error. Postmark didn't use to support this before but now according to [this](https://postmarkapp.com/blog/postmark-now-with-multiple-recipients-support) blog post from May 31st, 2019 it's now supported. Now that this is the case can we safely remove that from the backend?

**Comment by medmunds** [comment]
That assertion should only apply when you're trying to send a message with a `template_id` but haven't set `merge_data`. Is that what you're doing? (You want every recipient to receive exactly the same template message, without any per-recipient customizations, *and* you want them all to see each other's emails on the To line.)

With that combination of options, Anymail has decided it must use Postmark's [email/withTemplate API](https://postmarkapp.com/developer/api/templates-api#email-with-template). But it looks like Anymail is not handling this case properly for multiple recipients. It should probably check that the payload's `merge_data` and `merge_metadata` have not been set (otherwise it would use one of the batch APIs instead), and then just set `data = self.data`.

**Comment by medmunds** [comment]
Fix released in v8.3
